tests/helpers/test_template.py

```diff
@@ -582,6 +576,16 @@ def test_closest_function_invalid_coordinates(self):
                             '{{ closest("invalid", "coord", states) }}'))
 
     def test_closest_function_no_location_states(self):
-        """."""
+        """Test closest function without location states."""
         self.assertEqual('None',
                          template.render(self.hass, '{{ closest(states) }}'))
+
+    def test_compiling_template(self):
+        """Test compiling a template."""
+        self.hass.states.set('test_domain.hello', 'world')
+        compiled = template.compile_template(
+            self.hass, '{{ states.test_domain.hello.state }}')
+
+        with patch('homeassistant.helpers.template.compile_template',
+                   side_effect=Exception('Should not be called')):
+            assert 'world' == template.render(self.hass, compiled)
```

Comment:
Only suggesting since that is the format you use [here](https://github.com/home-assistant/home-assistant/pull/3515/files#diff-271c3f410a76f769b10ff2dd3c7ec7d7R69)